### PR TITLE
uploading recordings, saving settings, starting with services

### PR DESCRIPTION
**Pull request by laila-chammaa**

### CODE REVIEW

File: app/src/main/java/com/lla/panicbutton/ui/fragments/UploadFragment.kt [hunk 1]
```diff
@@ -39,24 +58,88 @@ class UploadFragment : Fragment(R.layout.fragment_upload) {
...
+        // we are creating the dir where we'll be storing the user's recordings and returning it to write to it
+        val recordingsDir = File(
+            context.getExternalFilesDir(
+                Environment.DIRECTORY_MUSIC
+            ), folderName
+        )
+        if (!recordingsDir.mkdirs()) {
+            Timber.d("${Constants.WARN_LOG_TAG} Directory not created")
+        }
+        return recordingsDir
```
Comment (laila-chammaa) [inline app/src/main/java/com/lla/panicbutton/ui/fragments/UploadFragment.kt]: @lisa7012 @Zandrenel could you help me with these two? haven't worked on this in a while, but need a second pair of eyes on these two. getting the length of audio file uploaded and creating a dir in the user's phone to store the audio they uploaded

Comment (laila-chammaa) [reply app/src/main/java/com/lla/panicbutton/ui/fragments/UploadFragment.kt]: you could directly pull the branch if u want

**Comment by laila-chammaa** [comment]
even tho it's not perfect and has some issues, merging now would be easier for conflict.

File: app/src/main/java/com/lla/panicbutton/ui/fragments/UploadFragment.kt [hunk 2]
```diff
@@ -128,8 +128,10 @@ class UploadFragment : Fragment(R.layout.fragment_upload) {
     private fun getAudioLengthMillis(context: Context, uri: Uri): Long {
         val mediaMetadataRetriever = MediaMetadataRetriever()
         mediaMetadataRetriever.setDataSource(context, uri)
-        return mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)!!
+        val length = mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)!!
             .toLong()
+        mediaMetadataRetriever.release()
+        return length
```
Comment (laila-chammaa) [inline app/src/main/java/com/lla/panicbutton/ui/fragments/UploadFragment.kt:134]: @lisa7012

**Review by lisa7012 — APPROVED** [review]
I think it's good to merge! Thank you!

**Pull request merged**